lua/neo-tree/sources/filesystem/init.lua

```diff
@@ -369,6 +369,15 @@ M.setup = function(config, global_config)
     })
   end
 
+  if global_config.default_component_configs.name.highlight_open_files then
```

Comment:
So in that case we'll have 2 configs  (1 global config and 1 for the name component) to turn this on?

Reply:
Yes, but I would default it to enabled since it's not really expensive the way it is handled. In that way, it's really one config to turn it on (the name component config) and one config to turn it off. The global config is really just fail safe for troubleshooting or people that want extreme optimization.

The alternative is to read the finalized merged config on every source and look for whether this feature is enabled, but that would be too complicated and then there's also the possibility that the data get's used for other reasons down the line.